extensions/microsoft-authentication/package.json

```diff
@@ -31,6 +31,36 @@
       {
         "label": "Microsoft",
         "id": "microsoft"
+      },
+      {
+        "label": "Azure Cloud",
+        "id": "azure-cloud"
+      }
+    ],
+    "configuration": [
+      {
+        "title": "Azure Cloud",
+        "properties": {
+          "azure-cloud.endpoint": {
+            "anyOf": [
+              {
+                "type": "string"
+              },
+              {
+                "type": "string",
+                "enum": [
+                  "https://login.chinacloudapi.cn/",
+                  "https://login.microsoftonline.us/"
+                ],
+                "enumDescriptions": [
+                  "Azure China",
+                  "Azure US Government"
+                ]
+              }
+            ],
+            "description": "%azure-cloud.endpoint.description%"
```

Comment:
Unfortunately, this throws out the Settings Editor: 
<img width="669" alt="image" src="https://user-images.githubusercontent.com/2644648/229226380-586b17f9-d366-4dc9-bd7b-04f747afedf7.png">

but I guess we don't have any alternative unless we do two settings (one for present, one for custom)

Reply:
Right, I was a bit bummed about that. I think that VSCode only offers a warning squiggle if you do a value outside the enum list--so, we could switch to the enum only, with the understanding that if you're using a custom cloud, you will get those squiggles and should ignore them.

Reply:
Opened https://github.com/microsoft/vscode/issues/178850